src/hooks/useFileUpload.tsx

```diff
@@ -246,12 +283,25 @@ export default function useFileUpload(
       })(body),
   });
 
-  const handleUpload = async (associating_id: string) => {
+  const handleUpload = async (
+    associating_id: string,
+    CombineToPDF?: boolean,
+  ) => {
```

Comment:
You should also modify allowedExtensions here based on CombineToPdf, before calling validateFileUpload.

Something like:

```
if (CombineToPDF && "allowedExtensions" in options) {
  options.allowedExtensions = ["jpg", "png", "jpeg"]
}
```

On a side note, rename CombineToPDF to combineToPDF.